rasa_core/channels/socketio.py

```diff
@@ -0,0 +1,144 @@
+        self.sio.emit(self.bot_message_evt, response, room=recipient_id)
+
+    def send_text_message(self, recipient_id, message):
+        # type: (Text, Text) -> None
+        """Send a message through this channel."""
+
+        self._send_message(recipient_id, {"text": message})
+
+    def send_image_url(self, recipient_id, image_url):
+        # type: (Text, Text) -> None
+        """Sends an image. Default will just post the url as a string."""
+        message = {
+            "attachment": {
+                "type": "image",
+                "payload": {"src": image_url}
+            }
+        }
+        self._send_message(recipient_id, message)
+
+    def send_text_with_buttons(self, recipient_id, text, buttons, **kwargs):
+        # type: (Text, Text, List[Dict[Text, Any]], **Any) -> None
+        """Sends buttons to the output."""
+
+        message = {
+            "text": text,
+            "quick_replies": []
+        }
+
+        for button in buttons:
+            message["quick_replies"].append({
+                "content_type": "text",
+                "title": button['title'],
+                "payload": button['payload']
+            })
+
+        self._send_message(recipient_id, message)
+
+    def send_custom_message(self, recipient_id, elements):
+        # type: (Text, List[Dict[Text, Any]]) -> None
+        """Sends elements to the output."""
+
+        message = {"attachment": {
+            "type": "template",
+            "payload": {
+                "template_type": "generic",
+                "elements": elements[0]
+            }}}
+
+        self._send_message(recipient_id, message)
+
+
+class SocketIOInput(InputChannel):
+    """A socket.io input channel."""
+
+    @classmethod
+    def name(cls):
+        return "socketio"
+
+    @classmethod
+    def from_credentials(cls, credentials):
+        credentials = credentials or {}
+        return cls(credentials.get("user_message_evt", "user_uttered"),
+                   credentials.get("bot_message_evt", "bot_uttered"),
+                   credentials.get("namespace"))
+
+    def __init__(self,
+                 user_message_evt="user_uttered",  # type: Text
+                 bot_message_evt="bot_uttered",  # type: Text
+                 namespace=None  # type: Optional[Text]
+                 ):
+        self.bot_message_evt = bot_message_evt
+        self.user_message_evt = user_message_evt
+        self.namespace = namespace
+
+    def blueprint(self, on_new_message):
+        sio = socketio.Server()
+        socketio_webhook = SocketBlueprint(sio, 'socketio_webhook', __name__)
+
+        @socketio_webhook.route("/", methods=['GET'])
+        def health():
+            return jsonify({"status": "ok"})
+
+        @sio.on('connect', namespace=self.namespace)
+        def connect(sid, environ):
+            logger.debug("User {} connected to socketio endpoint.".format(sid))
+
+        @sio.on('disconnect', namespace=self.name())
```

Comment:
why is this `self.name()` and the `connect` one  above `self.namespace`?

Reply:
good spot, should be namespace as well :+1: